modules/repository/init.go

```diff
@@ -45,18 +45,26 @@ var (
 func LoadRepoConfig() {
 	// Load .gitignore and license files and readme templates.
 	types := []string{"gitignore", "license", "readme", "label"}
-	typeFiles := make([][]string, 4)
+
+	removeExtension := func(f string) string {
+		ext := strings.ToLower(filepath.Ext(f))
+		if ext == ".yaml" || ext == ".yml" {
+			return f[:len(f)-len(ext)]
+		}
+		return f
+	}
```

Comment:
I do not understand why it should remove the extensions. It looks hacky and forces other code to "guess" the file name again and again.

And could there be a case like this:

* Gitea has builtin Advanced.yaml
* A user puts a customized Advance.yml
* Then still the builtin Advanced.yaml is used when guessing the name "Advanced"? IIRC the `options` package tries the custom directory first and then the builtin assets.

Reply:
I think it's completely reasonable to hide the extension - however, this does make overriding a little bit more complex.

Reply:
Or could the extension just be hidden on the UI? But always use the filename with extension internally?

Reply:
My proposal is  Respect the origin file names of option items #23749 

I think we do not need to guess.